core/kazoo_amqp/src/api/kapi.erl

```diff
@@ -57,3 +58,17 @@ encode_pid(Queue) ->
 -spec encode_pid(kz_term:ne_binary(), pid()) -> kz_term:ne_binary().
 encode_pid(Queue, Pid) ->
     list_to_binary(["pid://", kz_term:to_binary(Pid), "/", Queue]).
+
+-spec decode_pid(kz_term:ne_binary()) -> kz_term:api_pid().
+decode_pid(Queue) ->
+    case re:run(Queue, <<"pid://(.*)/(.*)">>, [{'capture', [1,2], 'binary'}]) of
+        {'match', [Pid, _RK]} -> kz_term:to_pid(Pid);
+        _Else -> 'undefined'
+    end.
+
+decode_pid2(<<"pid://", Pid/binary>>) ->
```

Comment:
@lazedo alternative implementation possibility

Reply:
see kz_api_tests perf test

Reply:
i like that. only change would be to make sure it would handle `<<"pid://<0.1.0>/2bdace1ec24371/1cc5236e01d5849d18/2bdace1ec243711cc5...">>`

Reply:
it does work as intended. we can use this instead. thanks